auth.php

```diff
@@ -266,6 +235,56 @@ public function can_be_manually_set() {
     public function signup_form() {
         return new \auth_enrolkey\form\enrolkey_signup_form(null, null, 'post', '', array('autocomplete' => 'on'));
     }
-}
 
+    /**
+     * @param bool $notify
+     * @param array $availableenrolids
+     * @param string $email
+     * @throws coding_exception
+     * @throws dml_exception
+     * @throws moodle_exception
+     */
+    private function enrolkey_notify(bool $notify, array $availableenrolids, string $email) {
+        global $PAGE, $OUTPUT, $CFG;
+        if ($notify === false) {
+            return;
+        }
+        if (get_config('auth_enrolkey', 'emailconfirmation')) {
```

Comment:
I would guess there is probably logic missing here now. If you have turned on email confirmation then you should get logged out if you are signing up, but you should not get kicked out if you have already confirmed your email and are using the block to enrol in a second course

Reply:
updated